packages/mdast-util-split/src/index.js

```diff
@@ -0,0 +1,124 @@
+module.exports = split
+import visit from 'unist-util-visit'
+
+function split (tree, {splitDepth = 1,
```

Comment:
`splitDepth` is not used.

Reply:
not yet implemented

Reply:
Now it's implemented, but I'm not happy at all with my namings, I always mix "trees" and "children" and it's quite complex when splitDepth is greater than 1.

Reply:
It's confusing I agree. I think `this.subTrees` actually refers to new root trees, not subtrees?

Try drawing a simple schema of the output for a simple md doc at various `splitDepth`, it should make it easier to see what is a root tree, what is a subtree (i.e. a subpart of mdast root tree)?

Reply:
```md
the introduction

# chapter 1

a paragraph

> a quote to *ensure this is parsed*

## section 1.1

content 1.1

## subsection 1.1.1

content 1.1.1

## section 1.2

content 1.2

# chapter 2

## subsection 2.1.1

content 2.1.1

# the conclusion

paragraph
```

For instance what is expected here with splitDepth 1 and splitDepth 2?

My guess for splitDepth 1:

```js
{
  introduction: {}, // root tree from doc root to `chapter 1` (not included)
  trees: [
      {}, // root tree from `chapter 1` to `chapter 2` (not included)
      {}, // root tree from `chapter 2` to `conclusion` (not included)
  ],
  conclusion: {}, // root tree from `chapter 2` (not included) to document end
}
```

Reply:
For this one, it is what I want.

Reply:
In fact, not, the `conclusion` must be from "conclusion" (not included)

Reply:
I guess this is what you mean:
```
{
  introduction: {type: root, children: [/* from doc root to `chapter 1` (not included) */]},
  trees: [
      {type: root, children: [/* from `chapter 1` to `chapter 2` (not included) */]},
      {type: root, children: [/* from `chapter 2` to `conclusion` (not included) */]},
  ],
  conclusion: {type: root, children: [/* from `conclusion` (not included) to document end */]},
}
```

What about splitDepth 2 for this document?

Reply:
this is what I mean. For splitDepth 2 here it what i would like : 

```
{
  introduction: {type: root, children: [/* from doc root to `chapter 1` (not included) */]},
  title: { type: root, children: [/*the heading node*/],
  trees: [
      {type: root, title: {type: 'root', children:[/*chapter 1 heading*/]},
        trees: [/** trees for section1.1=>section 1.2],
        conclusion: {type: root, children:[/* content of section 1.2, but heading discarded*/]},
      {type: root, children: [/* from `chapter 2` to `conclusion` (not included) */]},
  ],
  conclusion: {type: root, children: [/* from `conclusion` (not included) to document end */]},
}
```

Reply:
Why does chapter 1 have a conclusion but no introduction? Could you please explain how all of this works in the README?